test/e2e/framework/pod/pod_client.go

```diff
@@ -89,19 +90,32 @@ func PodClientNS(f *framework.Framework, namespace string) *PodClient {
 type PodClient struct {
 	f *framework.Framework
 	v1core.PodInterface
-	namespace string
+	namespace  string
+	trackOwner bool
+}
+
+// WithOwnerTracking enable automatic add of annotations recording the code location
+// which created a pod. This is helpful when troubleshooting e2e tests (like e2e_node)
+// which leak pods because insufficient cleanup.
+func (c *PodClient) WithOwnerTracking() *PodClient {
+	c.trackOwner = true
+	return c
```

Comment:
Would it make sense to have a per-test-suite default for this? This removes the need to touch >40 files and future E2E tests will do the right thing.

This method can have an explicit `enabled bool` parameter to override that default on a per-test basis.

I also would make it so that `WithOwnerTracking` returns a shallow clone.

Otherwise this code here won't work as expected:
```
c := e2epod.NewPodClient(f)
c.WithOwnerTracking(true).Create(...) // With tracking.
c.Create(...) // Also with tracking, which is unexpected.
```

You get that implicitly by changing the receiver to a value and returning a pointer to it.

Reply:
Yes, this is one the reasons why it's WIP. I'm looking for a smarter way to enable per-suite, but not obvious solutions yet, depends on how the code is laid out.

Re: shallow clone: very good idea, will do.

Reply:
Went for the "global enable in the suite". The approach grew on me and the code looks better than I thought.